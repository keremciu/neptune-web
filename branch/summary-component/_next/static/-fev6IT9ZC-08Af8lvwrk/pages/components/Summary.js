(window.webpackJsonp=window.webpackJsonp||[]).push([[52],{XHeJ:function(n,o,w){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/Summary",function(){return w("mcgz")}])}},[["XHeJ",0,1,4,5,3,2,6,7]]]);