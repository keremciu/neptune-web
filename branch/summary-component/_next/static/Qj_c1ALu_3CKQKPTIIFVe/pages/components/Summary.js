(window.webpackJsonp=window.webpackJsonp||[]).push([[52],{XHeJ:function(n,o,w){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/Summary",function(){return w("mcgz")}])}},[["XHeJ",0,1,4,5,2,3,6,7]]]);