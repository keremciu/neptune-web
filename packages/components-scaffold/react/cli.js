#!/usr/bin/env node

import prompt from 'prompt';
import path from 'path';
import fs from 'fs';
import colors from 'colors';
import templateComponent from './templates/component.js';
// import 'templateStyleFile' from "./templates/templateStyleFile");
import templateComponentExport from './templates/componentExport.js';
import templateJest from './templates/jest.js';
import templateStory from './templates/story.js';
import { capitalizeFirstLetter, message } from './utils/index.js';

const scrFolder = 'src';

//
// Start the prompt
//
prompt.start();

//
// Get the components infos
//
const schema = {
  properties: {
    name: {
      // pattern: /^[A-Za-z\s]+$/,
      // message: 'Please insert a component name: Only letters allowed',
      // required: true,
      default: 'test',
    },
    path: {
      message: 'Please insert a realtive component path: ',
      default: `${scrFolder}`,
    },
  },
};
const files = [
  {
    name: 'Index file',
    ext: 'js',
    template: templateComponentExport,
    fileName: 'index',
  },
  {
    name: 'Component file',
    ext: 'js',
    template: templateComponent,
  },
  // {
  //   name: "style file",
  //   ext: "less",
  //   template: templateStyleFile
  // },
  {
    name: 'Jest file',
    ext: 'spec.js',
    template: templateJest,
  },
  {
    name: 'Story file',
    ext: 'story.js',
    template: templateStory,
  },
];
prompt.get(schema, (err, result) => {
  const capitalizedName = capitalizeFirstLetter(result.name);

  const componentPath = `${process.env.INIT_CWD}/${result.path}/${result.name.toLowerCase()}`;

  if (!fs.existsSync(componentPath)) {
    try {
      fs.mkdirSync(componentPath);
      try {
        files.map((file) => {
          try {
            fs.writeFileSync(
              `${componentPath}/${file.fileName || capitalizedName}.${file.ext}`,
              file.template(capitalizedName),
            );
            console.log(message('success', file.name.green));
          } catch (err) {
            console.log(message('error', file.name.red, err));
          }
        });
      } catch (err) {
        console.log(message('error', 'directory').red, err);
      }
    } catch (err) {
      console.log('Error while creating directory'.red);
    }
  } else {
    console.log('Directory already exists'.red);
  }
});
