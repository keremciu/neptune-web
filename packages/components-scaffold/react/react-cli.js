#!/usr/bin/env node

const prompt = require('prompt');
const path = require('path');
const fs = require('fs');
const colors = require('colors');
const templateComponent = require('./templates/component');
// const templateStyleFile = require("./templates/templateStyleFile");
const templateIndex = require('./templates/componentExport');
const templateJest = require('./templates/jest');
const templateStory = require('./templates/story');
const { capitalizeFirstLetter, message } = require('./utils');

const scrFolder = 'src';

//
// Start the prompt
//
prompt.start();

//
// Get the components infos
//
const schema = {
  properties: {
    name: {
      // pattern: /^[A-Za-z\s]+$/,
      // message: 'Please insert a component name: Only letters allowed',
      // required: true,
      default: 'test',
    },
    path: {
      message: 'Please insert a realtive component path: ',
      default: `${scrFolder}`,
    },
  },
};
const files = [
  {
    name: 'Index file',
    ext: 'js',
    template: templateIndex,
    fileName: 'index',
  },
  {
    name: 'Component file',
    ext: 'js',
    template: templateComponent,
  },
  // {
  //   name: "style file",
  //   ext: "less",
  //   template: templateStyleFile
  // },
  {
    name: 'Jest file',
    ext: 'spec.js',
    template: templateJest,
  },
  {
    name: 'Story file',
    ext: 'story.js',
    template: templateStory,
  },
];
prompt.get(schema, (err, result) => {
  const capitalizedName = capitalizeFirstLetter(result.name);

  const componentPath = `${process.env.INIT_CWD}/${result.path}/${result.name.toLowerCase()}`;

  if (!fs.existsSync(componentPath)) {
    try {
      fs.mkdirSync(componentPath);
      try {
        files.map((file) => {
          try {
            fs.writeFileSync(
              `${componentPath}/${file.fileName || capitalizedName}.${file.ext}`,
              file.template(capitalizedName),
            );
            console.log(message('success', file.name.green));
          } catch (err) {
            console.log(message('error', file.name.red, err));
          }
        });
      } catch (err) {
        console.log(message('error', 'directory').red, err);
      }
    } catch (err) {
      console.log('Error while creating directory'.red);
    }
  } else {
    console.log('Directory already exists'.red);
  }
});
