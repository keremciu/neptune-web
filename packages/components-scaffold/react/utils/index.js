export const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1);
export const message = (type, message) => {
  if (type === 'error') {
    return `Failed to create ${message} file.`;
  }
  return `Component's ${message} created!`;
};
