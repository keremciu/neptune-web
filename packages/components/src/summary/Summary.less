@import '@transferwise/neptune-css/src/less/variables/_colors.less';
@import '@transferwise/neptune-css/src/less/variables/_spacing.less';
@import '@transferwise/neptune-css/src/less/variables/_typography.less';

.summary {
  &__main {
    display: flex;
  }

  &__icon {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-shrink: 0;
    width: 44px;
    height: 44px;
    margin-top: -10px;
    margin-bottom: -10px;
  }

  &__title {
    display: flex;
  }

  &__help {
    margin-top: 3px;
    height: 16px;

    .tw-icon {
      color: var(--brand-light-blue);

      width: 16px;
      height: 16px;
    }

    [data-toggle='popover']:not(.btn) {
      // Bootstrap popover has !important dotted border bottom
      // This doesn't make sense on a help icon
      border-bottom: none !important;
    }
  }

  &__content {
    margin-top: 4px;
    font-size: var(--font-size-small);
  }

  & + .summary {
    margin-top: calc(var(--spacer) * 3);
  }
}
