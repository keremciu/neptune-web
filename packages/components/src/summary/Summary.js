import classNames from 'classnames';
import React from 'react';
import Types from 'prop-types';
import { HelpCircle as HelpCircleIcon } from '@transferwise/icons';

import Popover from '../popover';

import './Summary.css';

const Summary = ({ as: Element, className, icon, title, help, content }) => (
  <Element className={classNames('summary', className)}>
    <div className="summary__main">
      <div className="summary__icon m-r-2">{icon}</div>
      <div className="summary__body">
        <div className="summary__title">
          <strong>{title}</strong>
          {help && (
            <div className="summary__help m-l-2">
              <Popover preferredPlacement="bottom-left" title={help.title} content={help.content}>
                <HelpCircleIcon />
              </Popover>
            </div>
          )}
        </div>
        {content && <div className="summary__content">{content}</div>}
      </div>
    </div>
  </Element>
);

Summary.propTypes = {
  as: Types.elementType,
  icon: Types.node.isRequired,
  title: Types.node.isRequired,
  content: Types.node,
  help: Types.shape({
    title: Types.node,
    content: Types.node.isRequired,
  }),
  className: Types.string,
};

Summary.defaultProps = {
  as: 'div',
  content: null,
  help: null,
  className: null,
};

export default Summary;
