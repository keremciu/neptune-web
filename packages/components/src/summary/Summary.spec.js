import React from 'react';
import { render } from '@testing-library/react';
import Summary from './Summary';

describe('Summary', () => {
  it('renders minimal component', () => {
    const { asFragment } = render(<Summary icon={<strong>icon</strong>} title="Hello world" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders full component', () => {
    const { asFragment } = render(
      <Summary
        icon={<strong>icon</strong>}
        title="Hello world"
        content={<p>This is some content</p>}
        help={{
          title: 'Need help?',
          content: 'This is some help content',
        }}
        className="outrageous"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
