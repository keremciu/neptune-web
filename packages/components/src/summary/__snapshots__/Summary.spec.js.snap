// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary renders full component 1`] = `
<DocumentFragment>
  <div
    class="summary outrageous"
  >
    <div
      class="summary__main"
    >
      <div
        class="summary__icon m-r-2"
      >
        <strong>
          icon
        </strong>
      </div>
      <div
        class="summary__body"
      >
        <div
          class="summary__title"
        >
          <strong>
            Hello world
          </strong>
          <div
            class="summary__help m-l-2"
          >
            <span
              data-toggle="popover"
              role="button"
              tabindex="0"
            >
              <span
                aria-hidden="true"
                class="tw-icon tw-icon-help-circle "
                role="presentation"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  width="16"
                >
                  <path
                    d="M9 11a1 1 0 11-2 0 1 1 0 012 0zM10.443 4.8A2.3 2.3 0 007.3 3.96L5.136 5.21l.8 1.385L8.1 5.345a.7.7 0 01.7 1.212l-2.165 1.25.8 1.386L9.6 7.943a2.3 2.3 0 00.842-3.142z"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M8 16A8 8 0 108 0a8 8 0 000 16zm0-1.6A6.4 6.4 0 108 1.6a6.4 6.4 0 000 12.8z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
            <div
              class="tw-popover popover animate in scale-down"
            >
              <h3
                class="popover-title"
              >
                Need help?
              </h3>
              <p
                class="popover-content m-b-0"
              >
                This is some help content
              </p>
            </div>
          </div>
        </div>
        <div
          class="summary__content"
        >
          <p>
            This is some content
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Summary renders minimal component 1`] = `
<DocumentFragment>
  <div
    class="summary"
  >
    <div
      class="summary__main"
    >
      <div
        class="summary__icon m-r-2"
      >
        <strong>
          icon
        </strong>
      </div>
      <div
        class="summary__body"
      >
        <div
          class="summary__title"
        >
          <strong>
            Hello world
          </strong>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
